### ISSUE TITLE
Add blanket file exclusion paths

### ISSUE THREAD

**Issue by TimMensch** [Microsoft/DevSkim-VSCode-Plugin#6]
I just tried the plug-in on my project, and it's giving me 251 errors in the auto-generated `yarn.lock`. 

```
file: 'file:///c%3A/Users/tim-m/projects/naming/synonymgraph/yarn.lock'
severity: 'Error'
message: '
Severity: [Important]

A token or key was found in source code. If this represents a secret, it should be moved somewhere else.

More Info:
https://github.com/Microsoft/DevSkim/blob/master/guidance/DS173237.md'
at: '7,74'
source: 'Devskim:Do not store tokens or keys in source code.'
```

Because it's auto-generated it can't be modified to disable a lint error, and so if DevSkim is enabled I have a huge number of "errors" that I can't eliminate in any way (that I can find in the docs).

My workaround is just to disable DevSkim. Let me know if there's a better way.

**Comment by scovetta** [comment]
I'm not terribly familiar with Yarn, but at a first glance, it looks like `yarn.lock` could always be ignored, since the contents are derived from `package.json`. We could have a global config value be a list of ignored glob expressions, defaulting to `["yarn.lock"]`, in which case none of the analysis would ever run on those files. 

The alternate would be more of a per-project `.devskim` configuration file that could contain things like that, and would travel around with the repo instead of being tied to a user's IDE config.

**Comment by TimMensch** [comment]
A list of glob expressions to ignore would be perfect. Defaulting it to block yarn.lock would be great too. 

And yes, yarn.lock is generated and as such shouldn't be security scanned. Would be fine to have a config file as well.

**Comment by joshbw** [comment]
I'll try to get an update out by the end of the week (baring unanticipated fire) that at least supports exclusions in the DevSkim VS Code settings, with some prepopulated excludes.  We'll work on having a unified exclusion .DevSkim file as well, but that may take a bit of synchronization in efforts between the three plugins

**Comment by joshbw** [comment]
Just added a setting to the VS Code version - https://github.com/Microsoft/DevSkim-VSCode-Plugin/commit/420fd53001e0d2aafc6afaa614433ec36de4ad5e

I have a couple other minor updates to make, and I need to fix one edge case with that update. will push a new version to the extension marketplace some time next week.

**Comment by TimMensch** [comment]
Awesome. Thanks!

**Comment by joshbw** [comment]
v 0.1.3 has been uploaded on the marketplace with this change.  Please let me know if the behavior is working correctly.  One caveat - if the file is already open with issues found, and you then ignore it, those issues will still be marked until the next time you open vs code (the inverse isn't true - if you opt to process a previously ignored file, the findings will show up without restarting VSC).  It appears that when returning an empty array of problems to VSC for a document it already has problems listed for, it doesn't then remove those problems.  I'm not sure why this is happening, and will take some debugging of VSC to figure out what it is actually doing.  Rather than have you wait while I did that I figured getting a "good enough" fix out there is probably best

**Comment by joshbw** [comment]
Quick update - I had to push out v0.1.4 since there was a JS exception during certain workplace update changes where VS code didn't always pass me the file path.  The upside is that I both fixed that, and my caveat above in 0.1.4 so things should be good to go.  Closing this issue out, though feel free to re-activate it if the implementation isn't what you were looking for
